src/modules/logicModule/paymentClient/ERC20PaymentClient.sol

```diff
@@ -208,4 +184,53 @@ abstract contract ERC20PaymentClient is
             revert Module__ERC20PaymentClient__InvalidRecipient();
         }
     }
+
+    //--------------------------------------------------------------------------
+    // {ERC20PaymentClient} Function Implementations
+
+    /// @dev Ensures `amount` of payment tokens exist in address(this).
+    function _ensureTokenBalance(uint amount) internal virtual {
+        uint balance = __Module_orchestrator.token().balanceOf(address(this));
```

Comment:
As discussed in the team meet earlier, we might do away with the pattern of storing the token address (IERC20) in the orchestrator contract in the future.

Possible reasons for this decision was to support multi-token payments and overall agreement that the proposal contract wasn't the best place to store the token address.

Therefore, I think querying the token from the orchestrator contract here, might not be a good move. I'd suggest querying this from the FundingManager contract.